### ISSUE TITLE
theme developments without reloading flask server

### ISSUE THREAD

**Issue by tomerp20** [CTFd/CTFd#2665]
**this is a continues issue for this one:**
https://github.com/CTFd/CTFd/issues/2607

**What happened?**

I attempted to improve the development workflow for CTFd by using the Vite dev server to serve asset files directly, instead of running vite build and restarting Flask for every change. While this works for serving files, I encountered issues related to Rollup pre-bundling. Specifically, when files are loaded unbundled during development, they don’t behave as expected, causing problems in the application.

**What did you expect to happen?**

I expected to use the Vite dev server to serve assets directly without encountering issues, improving the development workflow by avoiding the need to build and restart Flask after every file change.

**How to reproduce your issue**

Set up an environment variable, APP_ENV=dev, to toggle between development and production modes.
Modify the Assets class to serve JavaScript files directly from the Vite dev server in development mode:

```
if self.dev_mode:
    # In development mode, serve directly from Vite dev server
    entry = f"{self.vite_dev_server}/{asset_key}"
    return markup(f'<script type="module" src="{entry}"></script>')
```

Run the Vite dev server and load the app in the browser.
Observe the issues caused by files not being pre-bundled as they are in production.
Any associated stack traces or error logs

No stack traces, but the problem appears to stem from Rollup pre-bundling, which is not applied in the dev server workflow. The unbundled files cause errors in module resolution or execution.

**In conclusion**
I’d like to know if there’s a better solution to streamline the development process or if someone from the team can suggest how to handle these Rollup-related issues effectively.

Thanks for your time and help!

**Comment by ColdHeat** [comment]
This is an open issue as last I checked the vite server did not rebuild the manifest on rebuilds. CTFd needs the manifest.json file to determine where assets are so it's tough to use the dev server even though I would also like to use it. PR welcome if you can figure it out.
